hwy/tests/convert_test.cc

```diff
@@ -832,6 +832,227 @@ HWY_NOINLINE void TestAllFloatFromUint() {
+  static_assert(IsIntegerLaneType<TTo>() && IsSame<TTo, RemoveCvRef<TTo>>(),
+                "TTo must be an integer type");
+
+  template <class DF, HWY_IF_T_SIZE_LE_D(DF, sizeof(TTo) - 1)>
+  static HWY_INLINE VFromD<Rebind<TTo, DF>> DoF2IConvVec(DF /*df*/,
+                                                         VFromD<DF> v) {
+    return PromoteTo(Rebind<TTo, DF>(), v);
+  }
+
+  template <class DF, HWY_IF_T_SIZE_D(DF, sizeof(TTo))>
+  static HWY_INLINE VFromD<Rebind<TTo, DF>> DoF2IConvVec(DF /*df*/,
+                                                         VFromD<DF> v) {
+    return ConvertTo(Rebind<TTo, DF>(), v);
+  }
+
+  template <class DF, HWY_IF_T_SIZE_GT_D(DF, sizeof(TTo))>
+  static HWY_INLINE VFromD<Rebind<TTo, DF>> DoF2IConvVec(DF /*df*/,
+                                                         VFromD<DF> v) {
+    return DemoteTo(Rebind<TTo, DF>(), v);
+  }
+
+  template <class DF, HWY_IF_T_SIZE_LE_D(DF, sizeof(TTo) - 1)>
+  static HWY_INLINE Mask<Rebind<TTo, DF>> DoF2IConvMask(DF df, Mask<DF> m) {
+    return PromoteMaskTo(Rebind<TTo, DF>(), df, m);
+  }
+
+  template <class DF, HWY_IF_T_SIZE_D(DF, sizeof(TTo))>
+  static HWY_INLINE Mask<Rebind<TTo, DF>> DoF2IConvMask(DF /*df*/, Mask<DF> m) {
+    return RebindMask(Rebind<TTo, DF>(), m);
+  }
+
+  template <class DF, HWY_IF_T_SIZE_GT_D(DF, sizeof(TTo))>
+  static HWY_INLINE Mask<Rebind<TTo, DF>> DoF2IConvMask(DF df, Mask<DF> m) {
+    return DemoteMaskTo(Rebind<TTo, DF>(), df, m);
+  }
+
+  template <class DF, HWY_IF_T_SIZE_LE_D(DF, sizeof(TTo) - 1)>
+  static HWY_INLINE Vec<Rebind<MakeSigned<TTo>, DF>> DoF2IConvMsbMaskVec(
+      DF /*df*/, Vec<DF> v) {
+    return PromoteTo(Rebind<MakeSigned<TTo>, DF>(),
+                     BitCast(RebindToSigned<DF>(), v));
+  }
+
+  template <class DF, HWY_IF_T_SIZE_D(DF, sizeof(TTo))>
+  static HWY_INLINE Vec<Rebind<MakeSigned<TTo>, DF>> DoF2IConvMsbMaskVec(
+      DF /*df*/, Vec<DF> v) {
+    return BitCast(Rebind<MakeSigned<TTo>, DF>(), v);
+  }
+
+  template <class DF, HWY_IF_T_SIZE_GT_D(DF, sizeof(TTo))>
+  static HWY_INLINE Vec<Rebind<MakeSigned<TTo>, DF>> DoF2IConvMsbMaskVec(
+      DF /*df*/, Vec<DF> v) {
+    return DemoteTo(Rebind<MakeSigned<TTo>, DF>(),
+                    BitCast(RebindToSigned<DF>(), v));
+  }
+
+  template <class DF>
+  static HWY_NOINLINE void VerifyNonFiniteF2I(DF df, const VecArg<VFromD<DF>> v,
+                                              const char* filename,
+                                              const int line) {
+    using TF = TFromD<DF>;
+    using TU = MakeUnsigned<TF>;
+    using TTo_I = MakeSigned<TTo>;
+
+    constexpr TF kMinOutOfRangePosVal =
+        static_cast<TF>((-static_cast<TF>(LimitsMin<TTo_I>())) *
+                        static_cast<TF>(IsSigned<TTo>() ? 1 : 2));
+    static_assert(kMinOutOfRangePosVal > static_cast<TF>(0),
+                  "kMinOutOfRangePosVal > 0 must be true");
+
+    TU scalar_non_elided_zero = static_cast<TU>(0);
+#if HWY_COMPILER_MSVC
```

Comment:
Consider instead using Unpredictable1() from nanobenchmark.h? (Then subtract one from that to get zero)

Reply:
Already replaced the above code with `Unpredictable1() - 1`